### PR TITLE
feat: JSON output

### PR DESCRIPTION
**Pull request by Tiebe**
### What did you fix? (provide a description or issue closes statement)
Add json output

### CODE REVIEW

File: src/main/java/us/warframestat/moddetection/api/detection/DetectModInfo.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/detection/DetectModInfo.java:52]: null check on App.data in case it doesn't actually get defined (somehow?)

File: src/main/java/us/warframestat/moddetection/api/detection/DetectMods.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/detection/DetectMods.java:128]: javadoc?

File: src/test/java/us/warframestat/moddetection/api/AppTest.java
Comment (TobiTenno) [inline src/test/java/us/warframestat/moddetection/api/AppTest.java:37]: clean up App data after test?

**Comment by Tiebe** [comment]
I'll do those things, and I'll also add that you can specify the data directory where everything already is, so that the entire process is faster and doesn't take up a lot of space

**Comment by TobiTenno** [comment]
could you add the ci and verifications so we can check all of that stuff?

**Review by TobiTenno — CHANGES_REQUESTED** [review]
lookin for some base level ci so we can verify change to change

File: pom.xml
Comment (Bourne-ID) [inline pom.xml:9]: Non blocker: Want to keep to Semantic Versioning?

File: src/main/java/us/warframestat/moddetection/api/utils/WarframeMarketAPI.java
Comment (Bourne-ID) [inline src/main/java/us/warframestat/moddetection/api/utils/WarframeMarketAPI.java:108]: Logging here would still be useful, even if lower priority.  No-one wants to wait with no output on what's going on

File: src/test/java/us/warframestat/moddetection/api/AppTest.java
Comment (Bourne-ID) [inline src/test/java/us/warframestat/moddetection/api/AppTest.java:14]: Capitalisation of packages? Feel free to correct me...

File: src/test/java/us/warframestat/moddetection/api/AppTest.java
Comment (Bourne-ID) [inline src/test/java/us/warframestat/moddetection/api/AppTest.java:36]: Is there any risks of these loggers failing due to NullPointerExceptions? Are these calls guaranteed to be an object at minimum?

File: src/test/java/us/warframestat/moddetection/api/AppTest.java
Comment (Bourne-ID) [reply src/test/java/us/warframestat/moddetection/api/AppTest.java:14]: also: api.API?

**Comment by Tiebe** [comment]
Sorry, was busy the past two weeks, I'll fix some issues later today

File: pom.xml
Comment (TobiTenno) [reply pom.xml:9]: ideally, yeah, so next up should be 1.1.0... this should get incremented by the ci release, right?

File: src/main/java/us/warframestat/moddetection/api/utils/WarframeMarketAPI.java
Comment (TobiTenno) [reply src/main/java/us/warframestat/moddetection/api/utils/WarframeMarketAPI.java:108]: yeah, maybe do debug here instead of info?

File: src/test/java/us/warframestat/moddetection/api/AppTest.java
Comment (TobiTenno) [reply src/test/java/us/warframestat/moddetection/api/AppTest.java:14]: i think the idea is that .api is the last package, .API is the class, .LOGGER is an inner class, so it makes general sense

File: src/test/java/us/warframestat/moddetection/api/AppTest.java
Comment (TobiTenno) [reply src/test/java/us/warframestat/moddetection/api/AppTest.java:36]: thankfully it's in a test, but yeah, should probably do some asserts that they're not null

**Comment by TobiTenno** [comment]
i'm good with it, but i'll let @Bourne-ID have final approval since i made some changes

**Comment by Bourne-ID** [comment]
Will take a look once I’m back home, about 45 minutes.

File: .github/workflows/ci.yaml
Comment (Bourne-ID) [inline .github/workflows/ci.yaml:23]:
```suggestion
          MAVEN_USER: ${{ secrets.GIT_AUTHOR_NAME }}
          
```

File: .mvn/jvm.config
Comment (Bourne-ID) [inline .mvn/jvm.config:5]:
```suggestion
--add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAME

```

File: .mvn/settings.xml
Comment (Bourne-ID) [inline .mvn/settings.xml:13]:
```suggestion
</settings>

```

File: pom.xml
Comment (Bourne-ID) [inline pom.xml:74]: Moving away from variables here?

**Review by Bourne-ID — COMMENTED** [review]
Couple of minor comments but non-blockers on merge.

File: .github/workflows/ci.yaml
Comment (Bourne-ID) [reply .github/workflows/ci.yaml:23]: EOL

Comment (TobiTenno) [reply .github/workflows/ci.yaml:23]: I think I'm missing something. Is it switching the EOL or an EOF blank line?

File: pom.xml
Comment (TobiTenno) [reply pom.xml:74]: I think we probably want to keep the vars so we don't have to traverse as far for the config

File: src/main/java/us/warframestat/moddetection/api/API.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/API.java:19]: I just registered that this is public static. That is pretty problematic for it to be reset potentially whine it's being used

Should probably be private.

File: src/main/java/us/warframestat/moddetection/api/API.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/API.java:17]: I don't think consumers should have access to the internal logger

File: src/main/java/us/warframestat/moddetection/api/API.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/API.java:49]: We probably need to do this in startup on main to make sure we don't crash mid-process

File: src/main/java/us/warframestat/moddetection/api/API.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/API.java:61]: Cleanup should be internal and controlled

File: src/main/java/us/warframestat/moddetection/api/API.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/API.java:63]: While DATA and this method are public, that's a pretty big security risk

File: src/main/java/us/warframestat/moddetection/api/detection/DetectModInfo.java
Comment (TobiTenno) [inline src/main/java/us/warframestat/moddetection/api/detection/DetectModInfo.java:54]: In the future, let's look at preconstruction error JSON/GSON objects so we don't do literals that might get problematic later

**Comment by TobiTenno** [comment]
ok, so looks like i'm missing some linkage for it to work all the way

**Comment by TobiTenno** [comment]
@Bourne-ID @Tiebe what do y'all think of the changes?

**Comment by Tiebe** [comment]
> @Bourne-ID @Tiebe what do y'all think of the changes?

Sorry for the late reply, but looks good to me!

**Review by TobiTenno — APPROVED** [review]

**Pull request merged**
